core/doc_data.h

```diff
@@ -122,28 +122,36 @@ class DocData {
 		String keywords;
 		bool operator<(const MethodDoc &p_method) const {
 			if (name == p_method.name) {
-				// Must be an operator or a constructor since there is no other overloading
+				// Must be an operator or a constructor since there is no other overloading.
 				if (name.left(8) == "operator") {
 					if (arguments.size() == p_method.arguments.size()) {
-						if (arguments.size() == 0) {
+						if (arguments.is_empty()) {
 							return false;
 						}
 						return arguments[0].type < p_method.arguments[0].type;
 					}
 					return arguments.size() < p_method.arguments.size();
 				} else {
-					// Must be a constructor
+					// Must be a constructor.
 					// We want this arbitrary order for a class "Foo":
 					// - 1. Default constructor: Foo()
 					// - 2. Copy constructor: Foo(Foo)
-					// - 3+. Other constructors Foo(Bar, ...) based on first argument's name
-					if (arguments.size() == 0 || p_method.arguments.size() == 0) { // 1.
+					// - 3. Constructors starting with class: Foo(Foo, ...)
+					// - 4. Other constructors by arguments
+					if (arguments.is_empty() || p_method.arguments.is_empty()) { // 1.
 						return arguments.size() < p_method.arguments.size();
 					}
-					if (arguments[0].type == return_type || p_method.arguments[0].type == p_method.return_type) { // 2.
-						return (arguments[0].type == return_type) || (p_method.arguments[0].type != p_method.return_type);
+					if (arguments[0].type == return_type || p_method.arguments[0].type == return_type) { // 2 & 3 vs 4.
+						return p_method.arguments[0].type != p_method.return_type;
```

Comment:
The claimed sort above wasn't being done clearly, this enforces that (see below with `Array` and `Color` not following this order)